### PR TITLE
Upgrade to polymer 3

### LINKED ISSUE #12: Upgrade to Polymer 3

### PR DESCRIPTION
**Pull request by cristinecula**
Fixes #12

### CODE REVIEW

**Comment by cristinecula** [comment]
@megheaiulian @nomego this should be good to go :)

**Comment by cristinecula** [comment]
@nomego To be able to include this component in the other components, it needs to be published to npm. What package name do you want to use? Do you want to leave it as `cosmoz-moment` or "move" it under the organization `@neovici/cosmoz-moment`?

**Review by megheaiulian — APPROVED** [review]

**Comment by cristinecula** [comment]
I updated the package name to `@neovici/cosmoz-moment` and set the version to `3.0.0`.

File: cosmoz-moment-timeago.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+/**
+ * @polymer
+ * @customElement
+ * @extends PolymerElement
+ */
+class CosmozMomentTimeago extends mixinBehaviors([
+	Cosmoz.MomentBehavior
+], PolymerElement) {
+	static get template() {
+		return html`
```
Comment (nomego) [inline cosmoz-moment-timeago.js:14]: Why not a a single line ?

File: cosmoz-moment-timeago.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+`;
+	}
+
+	static get is() {
+		return 'cosmoz-moment-timeago';
+	}
+
+	static get observers() {
+		return [
+			'refreshIntervalChanged(refreshInterval)'
```
Comment (nomego) [inline cosmoz-moment-timeago.js:25]: Should be defined as an `observer` function within the property declaration.

File: cosmoz-moment-timeago.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+		};
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		window.clearInterval(this._intervalId);
+	}
+
+	refreshIntervalChanged(refreshInterval) {
+		this.set('_intervalId', window.setInterval(function () {
```
Comment (nomego) [inline cosmoz-moment-timeago.js]: Arrow function ?

File: cosmoz-moment.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+ */
+class CosmozMoment extends PolymerElement {
+
+	static get is() {
+		return 'cosmoz-moment';
+	}
+
+	static get observers() {
+		return [
+			'localeChanged(locale)'
```
Comment (nomego) [inline cosmoz-moment.js:94]: Should also be an observer function in the property declaration.

**Review by nomego — APPROVED** [review]
Can be merged as is but there are some stuff that should be captured for future changes.
(The PR really is about porting and I assume everything commented on is like this right now)
Also we should add a `Cosmoz.Mixins.moment()` mixin to use instead of using the behavior directly.

**Comment by cristinecula** [comment]
@nomego it appears that `polymer-analyzer` (which is used by `polymer lint`) expects that the observer descriptor is always a string. This makes the linter fail with observers defined in the way documented in https://polymer-library.polymer-project.org/2.0/docs/devguide/observers#simple-observers:

```
  static get properties() {
    return {
      active: {
        type: Boolean,
        // Observer method
        observer(newValue, oldValue){
          this.toggleClass('highlight', newValue);
        }
      }
    }
  }
```

If you go to the [3.0 documentation](https://polymer-library.polymer-project.org/3.0/docs/devguide/observers#simple-observers) you can see that the section has been removed.

**Comment by cristinecula** [comment]
@nomego ready to release!

**Pull request merged**